### PR TITLE
Rework mod to become a metapackage

### PR DESCRIPTION
**Pull request by rollerozxa**
The submodules contained within this modpack are quite outdated, so I reworked it to be a metapackage. When it is installed from ContentDB, it will install all of the mods alongside as dependencies.

### CODE REVIEW

File: mod.conf [hunk 1]
```diff
@@ -0,0 +1,4 @@
+name = castle
+title = Castles++
+description = This is a modpack all about creating castles and castle dungeons.
+depends = anvil, castle_farming, castle_gates, castle_lighting, castle_masonry, castle_shields, castle_storage, castle_tapestries, castle_weapons, crafting_bench orbs_of_time, ropes
```
Comment (SmallJoker) [inline mod.conf]: Missing separator after `crafting_bench`?

**Comment by Niklp09** [comment]
This makes https://github.com/minetest-mods/castle/issues/41, https://github.com/minetest-mods/castle/issues/35 and https://github.com/minetest-mods/castle/pull/39 obsolete.
Nice idea btw

**Review by Niklp09 — APPROVED** [review]
conditional approval on SmallJoker's review being addressed

File: mod.conf (hunk 1 above)
Comment (rollerozxa) [reply mod.conf]: oops, yes, sorry about that

**Comment by rollerozxa** [comment]
I also forgot to add an `init.lua` file since it's a mod now. Added an empty one in my fix commit alongside fixing the depends list.

**Comment by SmallJoker** [comment]
LGTM. Will merge in a few days unless there are objections.

**Pull request merged**